iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -720,4 +744,42 @@ private static FilesForCommit readFileForCommit(String fileForCommitLocation, Fi
       throw new NotFoundException("Can not read or parse committed file: %s", fileForCommitLocation);
     }
   }
+
+  public List<DataFile> getWrittenFiles(List<JobContext> jobContexts) throws IOException {
+    List<JobContext> jobContextList = jobContexts.stream()
+            .map(TezUtil::enrichContextWithVertexId)
+            .collect(Collectors.toList());
+    List<OutputTable> outputs = collectOutputs(jobContextList);
+    ExecutorService fileExecutor = fileExecutor(jobContextList.get(0).getJobConf());
+    ExecutorService tableExecutor = tableExecutor(jobContextList.get(0).getJobConf(), outputs.size());
+    Collection<DataFile> dataFiles = new ConcurrentLinkedQueue<>();
+    try {
+      Tasks.foreach(outputs.stream().flatMap(kv -> kv.jobContexts.stream()
+                      .map(jobContext -> new SimpleImmutableEntry<>(kv.table, jobContext))))
+              .suppressFailureWhenFinished()
+              .executeWith(tableExecutor)
+              .onFailure((output, exc) -> LOG.warn("Failed to retrieve merge input file for the table {}", output, exc))
+              .run(output -> {
+                JobContext jobContext = output.getValue();
+                JobConf jobConf = jobContext.getJobConf();
+                LOG.info("Cleaning job for jobID: {}, table: {}", jobContext.getJobID(), output);
+
+                Table table = output.getKey();
+                String jobLocation = generateJobLocation(table.location(), jobConf, jobContext.getJobID());
+                // list jobLocation to get number of forCommit files
+                // we do this because map/reduce num in jobConf is unreliable
+                // and we have no access to vertex status info
+                int numTasks = listForCommits(jobConf, jobLocation).size();
+                FilesForCommit results = collectResults(numTasks, fileExecutor, table.location(), jobContext,
+                        table.io(), false);
+                dataFiles.addAll(results.dataFiles());
+              }, IOException.class);
+    } finally {
+      fileExecutor.shutdown();
+      if (tableExecutor != null) {
```

Comment:
can it be null? if not, why don't we have same if for the fileExecutor

Reply:
`fileExecutor` is always created, whereas `tableExecutor` can be null.